### PR TITLE
Support existing SNS topics

### PR DESCRIPTION
**Pull request by AntonBazhal**
Sorry for the big number of changes. Had to change the most important styling problems, at least.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -93,25 +92,25 @@ class Plugin {
...
-					Ref: ref,
-				},
-			};
-                    });
 
-                    properties.Dimensions = dimensions
-                }
+		if (definition.pattern) {
+			//
+			properties.Namespace = this.serverless.service.service + '_' + properties.Namespace;
```
Comment (dougmoscrop) [inline src/index.js]: Could use backticks here.

File: src/index.js [hunk 2]
```diff
@@ -133,64 +132,66 @@ class Plugin {
...
+			const CFLogName = this.serverless.providers.aws.naming.getLogGroupLogicalId(functionName);
+			output[logMetricCFRef] = {
+				Type: "AWS::Logs::MetricFilter",
+				DependsOn: CFLogName,
+				Properties: {
+					FilterPattern: alarm.pattern,
+					LogGroupName: this.serverless.providers.aws.naming.getLogGroupName(this.serverless.service.getFunction(functionName).name),
+					MetricTransformations:[{
+						MetricValue: 1,
+						MetricNamespace:this.serverless.service.service + '_' + alarm.namespace,
```
Comment (dougmoscrop) [inline src/index.js]: Could switch to backticks here since it is done elsewhere.

File: src/index.js [hunk 3]
```diff
@@ -133,64 +132,66 @@ class Plugin {
...
+		if (alarm.pattern) {
+      //add custom log metric
+			const logMetricCFRef = this.naming.getLogMetricCFRef(normalizedFunctionName,alarm.name);
+			const CFLogName = this.serverless.providers.aws.naming.getLogGroupLogicalId(functionName);
+			output[logMetricCFRef] = {
+				Type: "AWS::Logs::MetricFilter",
+				DependsOn: CFLogName,
+				Properties: {
+					FilterPattern: alarm.pattern,
+					LogGroupName: this.serverless.providers.aws.naming.getLogGroupName(this.serverless.service.getFunction(functionName).name),
```
Comment (dougmoscrop) [inline src/index.js]: I think maybe use `serverless.getProvider('aws')` instead (more future-proof if they preserve this getter but don't keep .providers objecT)

File: src/index.js (hunk 1 above)
Comment (AntonBazhal) [inline src/index.js]: Not my code, but I can change

File: src/index.js (hunk 2 above)
Comment (AntonBazhal) [inline src/index.js]: Not my code as well

File: src/index.js (hunk 3 above)
Comment (AntonBazhal) [inline src/index.js]: Again, not my code, but will change

**Comment by AntonBazhal** [comment]
@dougmoscrop This repo is a shitshow. I fixed what you suggested, but I'm not going to fully enforce styling unless we decide to maintain our version of this plugin.

**Pull request merged**